### PR TITLE
scheduler: add timer for latch wait.

### PR DESCRIPTION
**Pull request by disksing**
@siddontang @zhangjinpeng1987

### CODE REVIEW

File: src/storage/txn/scheduler.rs [hunk 1]
```diff
@@ -566,10 +568,7 @@ impl Scheduler {
         let lock = self.gen_lock(&cmd);
         let ctx = RunningCtx::new(cid, cmd, lock, callback);
         self.insert_ctx(ctx);
-
-        if self.acquire_lock(cid) {
-            self.get_snapshot(cid);
-        }
+        self.wakeup_cmd(cid);
```
Comment (zhangjinpeng1987) [inline src/storage/txn/scheduler.rs]: what this means?

Comment (zhangjinpeng1987) [inline src/storage/txn/scheduler.rs]: but the semantics not equal

Comment (zhangjinpeng1987) [inline src/storage/txn/scheduler.rs]: maybe we should rename `wakeup_cmd` to `move_on_cmd`

**Comment by siddontang** [comment]
LGTM

File: src/storage/txn/scheduler.rs (hunk 1 above)
Comment (disksing) [inline src/storage/txn/scheduler.rs]: Seems not accurate either... @siddontang @ngaut Any suggestions?

Comment (siddontang) [inline src/storage/txn/scheduler.rs]: lock_and_get_snapshot?

**Comment by siddontang** [comment]
Ping @disksing

**Comment by zhangjinpeng1987** [comment]
LGTM

File: src/storage/txn/scheduler.rs [hunk 2]
```diff
@@ -710,12 +713,13 @@ impl Scheduler {
...
         for wcid in wakeup_list {
-            self.wakeup_cmd(wcid);
+            self.lock_and_get_snapshot(wcid);
         }
     }
 
-    /// Wakes up the next command for execution.
-    fn wakeup_cmd(&mut self, cid: u64) {
+    /// Tries to acquire all the necessary latches. If successful, initiates a get
+    /// snapshot for furthur processing.
```
Comment (queenypingcap) [inline src/storage/txn/scheduler.rs]: If successful, initiates a get snapshot for furthur processing. ->If all the necessary ltches are acquired, the method initiates a get snapshot operation for furthur processing.

**Comment by ngaut** [comment]
LGTM

**Pull request merged**